vcenter_operator/phelm.py

```diff
@@ -48,6 +52,21 @@ class DeploymentState(object):
     items = attr.ib(default=attr.Factory(OrderedDict))
     actions = attr.ib(default=attr.Factory(OrderedDict))
 
+    def render(self, scope, options):
+        try:
+            template_names = env.list_templates(
+                filter_func=lambda x: (x.startswith(scope)
+                                       and x.endswith('.yaml.j2')))
+            for template_name in template_names:
+                try:
+                    template = env.get_template(template_name)
+                    result = template.render(options)
+                    self.add(result)
+                except (TemplateError, YAMLError):
+                    LOG.exception("Failed to render %s", template_name)
```

Comment:
While this is the same as before, this might be sub-optimal code nonetheless in that it deletes the deployment if it cannot render the new template. We might want to devise a method to keep the old deployment if we cannot render the new one.

Reply:
Sounds sensible and doable, but do we agree to keep it out of scope of this change?

Reply:
Yes.

Reply:
Created #8 for this.